src/lib/Modal.svelte

```diff
@@ -0,0 +1,75 @@
+<script>
+    export let isOpen = false;
+    export let title = "Default Title";
+    export let body = "Default body text goes here.";
+
+    function closeModal() {
+        isOpen = false;
```

Comment:
Re 1.  This design leaves the state of openness of the modal in the parent of the modal.  We should bring the state of openness of each modal into the modal itself to minimize the amount of code necessary to create a modal.

Maybe `isOpen` could be state in Modal.svelte with a function that filps the state of `isOpen`.  When a link to each modal is clicked the state for that modal is flipped.  When the closing x is clicked the state is flipped again.  This will take some thought.